devops/providers/azure-devops/templates/azure-pipelines.yml

```diff
@@ -61,10 +61,11 @@ steps:
     BASE_IMAGE=$ACR_HOST/$TEST_HARNESS_BASE_IMAGE:$TEST_HARNESS_BASE_IMAGE_TAG
     substitution="remotes/origin"
     replace_string="refs/heads"
+    GIT_DIFF_UPSTREAMBRANCH="remotes/origin/master"
     GIT_DIFF_SOURCEBRANCH=`[[ -n $SYSTEM_PULLREQUEST_SOURCEBRANCH ]] && \
         echo "${substitution}/${SYSTEM_PULLREQUEST_SOURCEBRANCH}" || \
         echo "${BUILD_SOURCEBRANCH//$replace_string/$substitution}"`
-    GIT_DIFF_UPSTREAMBRANCH="remotes/origin/master"
+    [ $GIT_DIFF_SOURCEBRANCH == $GIT_DIFF_UPSTREAMBRANCH ] && { GIT_DIFF_SOURCEBRANCH="$GIT_DIFF_SOURCEBRANCH~"; }
```

Comment:
Reading up on this, I suspect that this strategy will only work for squash merge PR. This is because in a squash merge PR, the whole commit history is converted into a single commit, so you can be sure that `GIT_DIFF_SOURCEBRANCH` is indeed `master~`.

If a non-squash merge is used, then each individual commit that the requester made will be added, so you don't know how many commits to look back when determining `GIT_DIFF_SOURCEBRANCH`. 

More info here (note, the images are extremely unclear here... read the text as it is more relevant.):
https://help.github.com/en/articles/about-pull-request-merges

Are we enforcing by some programmatic method that PRs are squash merge PRs?

Reply:
cobalt has enabled [this ](https://help.github.com/en/articles/configuring-commit-squashing-for-pull-requests) which forces developers to squash merges

Reply:
Awesome!